addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiAqaraSensorMagnetHandler.java

```diff
@@ -0,0 +1,97 @@
+package org.openhab.binding.mihome.handler;
+
+import static org.openhab.binding.mihome.XiaomiGatewayBindingConstants.*;
+
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+
+public class XiaomiAqaraSensorMagnetHandler extends XiaomiSensorBaseHandlerWithTimer {
+
+    private static final int DEFAULT_TIMER = 300;
+    private static final int MIN_TIMER = 30;
+    private static final String VOLTAGE = "voltage";
+    private static final String OPEN = "open";
+    private static final String CLOSED = "closed";
+    private static final String STATUS = "status";
+
+    private final Logger logger = LoggerFactory.getLogger(XiaomiSensorMagnetHandler.class);
+
+    public XiaomiAqaraSensorMagnetHandler(Thing thing) {
+        super(thing, DEFAULT_TIMER, MIN_TIMER, CHANNEL_OPEN_ALARM_TIMER);
+    }
+
+    @Override
+    void execute(ChannelUID channelUID, Command command) {
+        if (CHANNEL_OPEN_ALARM_TIMER.equals(channelUID.getId())) {
+            if (command != null && command instanceof DecimalType) {
+                setTimerFromDecimalType((DecimalType) command);
+                return;
+            }
+            // Only gets here, if no condition was met
+            logger.error("Can't handle command {} on channel {}", command, channelUID);
+        } else {
+            logger.error("Channel {} does not exist", channelUID);
+        }
+    }
+
+    @Override
+    void onTimer() {
+        triggerChannel(CHANNEL_OPEN_ALARM, "ALARM");
+    }
+
+    @Override
+    void parseReport(JsonObject data) {
+        parseDefault(data);
+    }
+
+    @Override
+    void parseHeartbeat(JsonObject data) {
+        parseDefault(data);
+    }
+
+    @Override
+    void parseReadAck(JsonObject data) {
+        parseDefault(data);
+    }
+
+    @Override
+    void parseWriteAck(JsonObject data) {
+        parseDefault(data);
+    }
+
+    @Override
+    void parseDefault(JsonObject data) {
```

Comment:
please check if sensor tells the truth in heartbeat messages for status & provide debug log